run-custom-mao.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -xe
+
+export SCRIPTS_DIR=$(pwd)
+export CUSTOM_MAO_BUILD=quay.io/sdasu/machine-api-operator:sdasu-fix
+
+kubectl get secrets --namespace openshift-machine-api | grep mariadb
+
+# Stop CVO, MAO, machine-api-controller and Metal3 deployments
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 cluster-version-operator --namespace openshift-cluster-version
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment -n openshift-machine-api --replicas=0 machine-api-operator
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 machine-api-controllers --namespace openshift-machine-api
+oc --config /home/stack/dev-scripts/ocp/auth/kubeconfig scale deployment --replicas=0 metal3 --namespace openshift-machine-api
+
+REPLACEMENT_TEXT="image: ${CUSTOM_MAO_BUILD}"
+echo $REPLACEMENT_TEXT
+
+oc get deployments -n openshift-machine-api machine-api-operator -o yaml > mao.yaml
+sed -i -e "s!image: .*!${REPLACEMENT_TEXT}!g" mao.yaml
+oc apply -f mao.yaml
+
+# Give the machine-api-operator rbac access to create and read secrets
+oc apply -f machine-api-operator-roles.yaml
```

Comment:
Are these roles different from the roles that will be set up by whatever deployed the MAO as part of its initial installation?

Reply:
Yes, the initial deployment does not give machine-api-operator permission to read, create & delete secrets etc. The MAO changes for creating its own mariadb password does contain an update to these roles but is not re-applied when we just modify the name of the machine-api-operator image and keep the cluster-version-operator scaled to 0.

Once https://github.com/openshift/machine-api-operator/pull/389 merges, we will no longer need machine-api-operator-roles.yaml to be part of dev-scripts (atleast until another change modifies this file).

Reply:
OK, good.